src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -250,22 +369,43 @@ export default class SpawnPatternModel extends BaseModel {
      * @method init
      * @param spawnPatternJson {object}
      * @param navigationLibrary {NavigationLibrary}
+     * @param airportController {AirportController}
      */
-    init(spawnPatternJson, navigationLibrary) {
+    init(spawnPatternJson, navigationLibrary, airportController) {
+        // This is a pooled object, so we still want internal properties to initialize first before returning.
+        if (!_isObject(spawnPatternJson) || _isEmpty(spawnPatternJson)) {
+            return;
+        }
+
+        if (!_isObject(navigationLibrary) || _isEmpty(navigationLibrary)) {
+            throw new TypeError('Invalid NavigationLibrary passed to SpawnPatternModel');
+        }
+
+        if (!_isObject(airportController) || _isEmpty(airportController)) {
+            throw new TypeError('Invalid AirportController passed to SpawnPatternModel');
+        }
+
+        this._airportController = airportController;
+
         this.origin = spawnPatternJson.origin;
         this.destination = spawnPatternJson.destination;
         this.category = spawnPatternJson.category;
         this.method = spawnPatternJson.method;
         this.rate = spawnPatternJson.rate;
         this.routeString = spawnPatternJson.route;
+        this.cycleStartTime = 0;
+        this.period = TIME.ONE_HOUR_IN_SECONDS / 2;
         this.speed = this._extractSpeedFromJson(spawnPatternJson);
         this._minimumDelay = this._calculateMinimumDelayFromSpeed();
         this._maximumDelay = this._calculateMaximumDelayFromRate();
```

Comment:
funny, I hadn't even noticed the similarity in method names.  Thats an advantage of a review, you'll see these things.

Obviously, the end of the method name is coming from a property in the `spawnPattern`, perhaps I change the name to `_calculateMaximumDelayFromRate` so the distinction will be a little clearer.